metrics/honeycomb.go

```diff
@@ -283,75 +278,89 @@ func average(vals []float64) float64 {
 func (h *HoneycombMetrics) Register(name string, metricType string) {
 	switch metricType {
 	case "counter":
-		h.countersLock.Lock()
-		defer h.countersLock.Unlock()
-		// inside the lock, let's not race to create the counter
-		_, ok := h.counters[name]
-		if !ok {
-			newCounter := &counter{
-				name: name,
-			}
-			h.counters[name] = newCounter
-		}
+		getOrAdd(&h.lock, name, h.counters, createCounter)
 	case "gauge":
-		h.gaugesLock.Lock()
-		defer h.gaugesLock.Unlock()
-		_, ok := h.gauges[name]
-		if !ok {
-			newGauge := &gauge{
-				name: name,
-			}
-			h.gauges[name] = newGauge
-		}
+		getOrAdd(&h.lock, name, h.gauges, createGague)
```

Comment:
```suggestion
		getOrAdd(&h.lock, name, h.gauges, createGauge)
```
(with function rename)